docs/concepts/README.md

```diff
@@ -0,0 +1,217 @@
+# Duck Typing
+
+Knative uses duck typing to keep various components loosely coupled, but
+[what is duck typing](https://en.wikipedia.org/wiki/Duck_typing)? It is
+reasoning about a resource's control plane shape and behaviors based on a common
+definition of that contact. If a resource has the same fields in the same schema
+locations as the contract specifies, and the control/data plane behaviors as the
+contract specifies, Knative can use that resource as if it is the generic duck
+type with little specific knowledge about the resource type. Some resources may
+choose to opt-in or be opt'ed-in to multiple duck types.
```

Comment:
```suggestion
type, without specific knowledge about the resource type. Some resources may
choose to opt-in to multiple duck types.
```

Reply:
I'd explain what it means to "opt in" better - does this just mean that they can use properties that would make them acceptable as more than one duck type?

Reply:
it means I can make a third party resource into a duck type (if it conforms to the shape) without the third party realizing this. the double hop integration